### PR TITLE
Activity Log Store - Part 5 - Rewind

### PR DESCRIPTION
**Pull request by planarvoid**
- implement rewind operation
- rewind is still on V1 rest endpoint - https://betterjetpackxp.wordpress.com/2017/10/05/activity-log-on-the-mobile-apps/
- it is not possible to reliably test rewind with real API so I'm just testing the failure (and that the endpoint is there) and unit testing the rest

### CODE REVIEW

File: fluxc/src/main/java/org/wordpress/android/fluxc/store/ActivityLogStore.kt [hunk 1]
```diff
@@ -78,6 +84,14 @@ class ActivityLogStore
         }
     }
 
+    private fun emitRewindResult(payload: RewindResultPayload, action: ActivityLogAction) {
+        if (payload.restoreId != null) {
+            emitChange(OnRewind(restoreId = payload.restoreId, causeOfChange = action))
+        } else if (payload.error != null) {
+            emitChange(OnRewind(payload.error, action))
+        }
```
Comment (aforcier) [inline fluxc/src/main/java/org/wordpress/android/fluxc/store/ActivityLogStore.kt]: Perhaps a null `restoreId` should be caught in the rest client and floated up here as a new error, `MISSING_RESTORE_ID` or something like that?

**Comment by aforcier** [comment]
Looks good to me overall, just had one comment/question - will let @0nko verify the functionality.

File: fluxc/src/main/java/org/wordpress/android/fluxc/store/ActivityLogStore.kt (hunk 1 above)
Comment (planarvoid) [reply fluxc/src/main/java/org/wordpress/android/fluxc/store/ActivityLogStore.kt]: good point. I should revert this logic as I did in AL

**Comment by 0nko** [comment]
:shipit:

**Pull request merged**